src/Filters/atoms/DateTimeBlock.tsx

```diff
@@ -143,12 +165,24 @@ const DateTimeBlock : React.FC<IProps> = ({
     if(allowAfterMidnight && isEqual(date, endOfDay(date))){
       setDisplayHours('24');
       setDisplayMinutes('00');
-    } else {
-      setDisplayMinutes(format(date, 'mm'));
-      setDisplayHours(format(date,'HH'));
     }
   },[date, allowAfterMidnight]);
 
+  const onBlurInputs = useCallback(() =>{
+    if(displayHours === '0' || displayHours === ''){
+      setDisplayHours('00');
+    }
+    if(displayMinutes ==='0' || displayMinutes ===''){
+      setDisplayMinutes('00');
+    }
+    if(displayMinutes.length === 1){
+      setDisplayMinutes('0' + displayMinutes);
+    }
+    if(displayHours.length === 1){
+      setDisplayHours('0' + displayHours);
+    }
```

Comment:
As mentioned in the previous code comment, it would be better to handle these in a unified function so it's easier to read and update later if needed.

Reply:
Done